File: lclang/src/main/java/lclang/statements/expressions/VariableExpression.java
package lclang.statements.expressions;

import lclang.Caller;
import lclang.LCBaseExecutor;
import lclang.Link;
import lclang.exceptions.LCLangRuntimeException;

public class VariableExpression extends Expression {
    public final String name;

    public VariableExpression(String name, int line) {
        super(line);
        this.name = name;
    }

    @Override
    public Link visit(Caller prevCaller, LCBaseExecutor visitor) throws LCLangRuntimeException {
        return visitor.getLink(getCaller(prevCaller), name);
    }

//    @Override
//    public List<Byte> compile(Caller prevCaller) {
//        ArrayList<Byte> bytes = new ArrayList<>();
//        bytes.add(Opcodes.GET);
//        bytes.addAll(new ArrayList<>());
//
//        List<Byte>
//        return bytes;
//    }
}

File: lclang/src/main/kotlin/lclang/statements/VariableExpression.kt
package lclang.statements

import lclang.Caller
import lclang.LCBaseExecutor

class VariableExpression(
    val name: String,
    line: Int,
    column: Int
): Expression(line, column) {
    override fun visit(prevCaller: Caller, visitor: LCBaseExecutor) = visitor.getLink(name)
}